WeakAuras/DebugLog.lua

```diff
@@ -0,0 +1,97 @@
+if not WeakAuras.IsCorrectVersion() or not WeakAuras.IsLibsOK() then return end
+local AddonName, Private = ...
+
+local WeakAuras = WeakAuras
+local L = WeakAuras.L
+
+local debugLogs = {}
+local enabled = {}
+
+Private.DebugLog = {
+}
+
+local function serialize(log, input)
+  if type(input) == "table" then
+    if log[#log] == "" then
+      log[#log] = L["Dumping table"]
+    else
+      tinsert(log, L["Dumping table"])
+    end
+    -- Use dump to create a table dump, because that already handles depth limitation
+    -- and cycles and looks nice
+    -- But this requires temporarily setting DEFAULT_CHAT_FRAME
+    -- Nothing can go wrong with that.
+    local defaultChatFrame = _G.DEFAULT_CHAT_FRAME
+    _G.DEFAULT_CHAT_FRAME = log
+    DevTools_Dump(input)
+    _G.DEFAULT_CHAT_FRAME = defaultChatFrame
+    tinsert(log, "")
+  else
+    if log[#log] == "" then
+      log[#log] = tostring(input)
+    else
+      log[#log] = log[#log] .. " " .. tostring(input)
+    end
+  end
+end
+
+function Private.DebugLog.Print(uid, text, ...)
+  if enabled[uid] then
+    local log = debugLogs[uid]
+    tinsert(log, "")
+    if select('#', ...) == 0 then
+      serialize(log, text)
+    else
+      serialize(log, text)
+      local texts = {...}
+      for i, v in ipairs(texts) do
```

Comment:
If one of the arguments is nil furthers arguments are ignored
`DebugPrint("a", "b", nil, "c", "d")`
will add in log  `a b`